src/bp-core/compatibility/bp-incompatible-plugins-helper.php

```diff
@@ -155,12 +158,210 @@ function bp_core_instructor_role_post_exclude( $exclude_posts ) {
 
 add_action( 'init', 'bp_helper_plugins_loaded_callback', 0 );
 
+/**
+ * Function to set the false to use the default media symlink instead use the offload media URL of media.
+ *
+ * @param bool   $can           default true.
+ * @param int    $id            media/document/video id.
+ * @param int    $attachment_id attachment id.
+ * @param string $size          preview size.
+ *
+ * @return bool true if the offload media used.
+ *
+ * @since BuddyBoss 1.7.0
+ */
+function bb_offload_do_symlink( $can, $id, $attachment_id, $size ) {
+	if ( class_exists( 'WP_Offload_Media_Autoloader' ) && class_exists( 'Amazon_S3_And_CloudFront' ) ) {
+		$remove_local_files_setting = bp_get_option( Amazon_S3_And_CloudFront::SETTINGS_KEY );
+		if ( isset( $remove_local_files_setting ) && isset( $remove_local_files_setting['bucket'] ) && isset( $remove_local_files_setting['copy-to-s3'] ) && '1' === $remove_local_files_setting['copy-to-s3'] ) {
+			$can = false;
+		}
+	}
+	return $can;
+}
+add_filter( 'bb_media_do_symlink', 'bb_offload_do_symlink', PHP_INT_MAX, 4 );
+add_filter( 'bb_document_do_symlink', 'bb_offload_do_symlink', PHP_INT_MAX, 4 );
+add_filter( 'bb_video_do_symlink', 'bb_offload_do_symlink', PHP_INT_MAX, 4 );
+add_filter( 'bb_video_create_thumb_symlinks', 'bb_offload_do_symlink', PHP_INT_MAX, 4 );
+
+/**
+ * Copy to local media file when the offload media used and remove local file setting used in offload media plugin to regenerate the thumb of the PDF.
+ *
+ * @param bool               $default default false.
+ * @param string             $file file to download.
+ * @param int                $attachment_id attachment id.
+ * @param Media_Library_Item $as3cf_item media library object.
+ *
+ * @return bool
+ *
+ * @since BuddyBoss 1.7.0
+ */
+function bb_document_as3cf_get_attached_file_copy_back_to_local( $default, $file, $attachment_id, $as3cf_item ) {
+	$default = true;
+	return $default;
+}
+
+/**
+ * Regenerate the the media attachments.
+ *
+ * @param int $attachment_id attachment id to recreate the media attachment.
+ *
+ * @since BuddyBoss 1.7.0
+ */
+function bb_document_wp_offload_regenerate_pdf_metadata( $attachment_id ) {
```

Comment:
@chetansatasiya I think this is unused function. Please check and remove if not used.

Reply:
done